lib/collection/src/collection/mod.rs

```diff
@@ -532,6 +513,21 @@ impl Collection {
             // Try to find dead replicas with no active transfers
             let transfers = shard_holder.get_transfers(|_| true);
 
+            // Respect shard transfer limit, consider already proposed transfers in our counts
+            let transfer_io = shard_holder.count_shard_transfer_io(this_peer_id);
+            let transfer_io = (
+                transfer_io.0 + *proposed.get(this_peer_id).unwrap_or(&0),
+                transfer_io.1,
+            );
+            if self.check_shard_transfer_limit(transfer_io) {
+                log::trace!(
+                    "Postponing automatic shard {shard_id} transfer to stay below limit on this node (i/o: {}/{})",
+                    transfer_io.0,
+                    transfer_io.1,
+                );
+                continue;
+            }
```

Comment:
I'm sorry, but this is unreadable.

I'm still confused regarding incoming/outgoing:
- `count_shard_transfer_io` returns `(outgoing, incoming)`
- but `check_shard_transfer_limit` expects `(incoming, outgoing)`
- but when we do `log::trace` we also use them "reversed":
  - `transfer_io.0` is `incoming`
  - and `transfer_io.1` is `outgoing`

I've made a few suggestions, without changing the order anywhere, but please clarify these. 😂

```suggestion
            let (mut outgoing, incoming = shard_holder.count_shard_transfer_io(this_peer_id);
            outgoing += proposed.get(this_peer_id).copied().unwrap_or(0);

            if self.check_shard_transfer_limit(incoming, outgoing) {
                log::trace!(
                    "Postponing automatic shard {shard_id} transfer to stay below limit on this node (i/o: {outgoing}/{incoming})"
                );
                continue;
            }
```

Reply:
It looks like I did mix up variables, but the result is the same.

The intent is to always do input/output because of io.